scripts/init_chain.sh

```diff
@@ -0,0 +1,177 @@
+
+    # Add interchain accounts to the genesis set
+    jq "del(.app_state.interchain_accounts)" $genesis_config > json.tmp && mv json.tmp $genesis_config
+    interchain_accts=$(cat $SCRIPT_DIR/config/ica.json)
+    jq ".app_state += $interchain_accts" $genesis_config > json.tmp && mv json.tmp $genesis_config
+}
+
+MAIN_ID=1 # Node responsible for genesis and persistent_peers
+MAIN_NODE_NAME=""
+MAIN_NODE_CMD=""
+MAIN_NODE_ID=""
+MAIN_CONFIG=""
+MAIN_GENESIS=""
+echo "Initializing $CHAIN_ID chain..."
+for (( i=1; i <= $NUM_NODES; i++ )); do
+    # Node names will be of the form: "stride-node1"
+    node_name="${NODE_PREFIX}${i}"
+    # Moniker is of the form: STRIDE_1
+    moniker=$(printf "${NODE_PREFIX}_${i}" | awk '{ print toupper($0) }')
+
+    # Create a state directory for the current node and initialize the chain
+    mkdir -p $STATE/$node_name
+    cmd="$CMD --home ${STATE}/$node_name"
+    $cmd init $moniker --chain-id $CHAIN_ID --overwrite &> /dev/null
+
+    # Update node networking configuration 
+    config_toml="${STATE}/${node_name}/config/config.toml"
+    client_toml="${STATE}/${node_name}/config/client.toml"
+    app_toml="${STATE}/${node_name}/config/app.toml"
+    genesis_json="${STATE}/${node_name}/config/genesis.json"
+
+    sed -i -E "s|cors_allowed_origins = \[\]|cors_allowed_origins = [\"\*\"]|g" $config_toml
+    sed -i -E "s|127.0.0.1|0.0.0.0|g" $config_toml
+    sed -i -E "s|timeout_commit = \"5s\"|timeout_commit = \"${BLOCK_TIME}\"|g" $config_toml
+    sed -i -E "s|prometheus = false|prometheus = true|g" $config_toml
+
+    sed -i -E "s|minimum-gas-prices = \".*\"|minimum-gas-prices = \"0${DENOM}\"|g" $app_toml
+    sed -i -E '/\[api\]/,/^enable = .*$/ s/^enable = .*$/enable = true/' $app_toml
+    sed -i -E 's|unsafe-cors = .*|unsafe-cors = true|g' $app_toml
+
+    sed -i -E "s|chain-id = \"\"|chain-id = \"${CHAIN_ID}\"|g" $client_toml
+    sed -i -E "s|keyring-backend = \"os\"|keyring-backend = \"test\"|g" $client_toml
+    sed -i -E "s|node = \".*\"|node = \"tcp://localhost:$RPC_PORT\"|g" $client_toml
+
+    sed -i -E "s|\"stake\"|\"${DENOM}\"|g" $genesis_json
+
+    # Get the endpoint and node ID
+    node_id=$($cmd tendermint show-node-id)@$node_name:$PEER_PORT
+    echo "Node #$i ID: $node_id"
+
+    # add a validator account
+    val_acct="${VAL_PREFIX}${i}"
+    val_mnemonic="${VAL_MNEMONICS[((i-1))]}"
+    echo "$val_mnemonic" | $cmd keys add $val_acct --recover --keyring-backend=test >> $KEYS_LOGS 2>&1
+    val_addr=$($cmd keys show $val_acct --keyring-backend test -a)
+    # Add this account to the current node
+    $cmd add-genesis-account ${val_addr} ${VAL_TOKENS}${DENOM}
+    # actually set this account as a validator on the current node 
+    $cmd gentx $val_acct ${STAKE_TOKENS}${DENOM} --chain-id $CHAIN_ID --keyring-backend test &> /dev/null
+
+    # Cleanup from seds
```

Comment:
thanks for adding this! i'm still not sure why the "-E" files get created in the first place

Reply:
I think it's just a mac thing unfortunately